controllers/capabilities/developeruser_threescale_reconciler.go

```diff
@@ -57,11 +60,15 @@ func (s *DeveloperUserThreescaleReconciler) Reconcile() (*threescaleapi.Develope
 
 	if devUser == nil {
 		s.logger.V(1).Info("DeveloperUser does not exist", "username", s.userCR.Spec.Username)
-		// developer user has to be created in 3scale
+		// The DeveloperAccount doesn't exist yet and must be created in 3scale
 		devUser, err = s.createDevUser()
 		if err != nil {
 			return nil, err
 		}
+
+		// Update the CR status with the account's ID and return the account
+		s.userCR.Status.ID = devUser.Element.ID
```

Comment:
Out of interest, why do we now need to do this? How was this handled before the changes?

Reply:
This wasn't handled properly before the changes and was part of the reason we were getting issues with the 409 error. If we don't include the line then the userCR doesn't have the correct ID which can lead to duplicate entries in the DB.